src/menu/cart_load.c

```diff
@@ -79,13 +79,55 @@ cart_load_err_t cart_load_n64_rom_and_save (menu_t *menu, flashcart_progress_cal
         return CART_LOAD_ERR_ROM_LOAD_FAIL;
     }
 
-    path_ext_replace(path, "sav");
-    if (menu->settings.use_saves_folder) {
-        if ((save_type != FLASHCART_SAVE_TYPE_NONE) && create_saves_subdirectory(path)) {
-            path_free(path);
-            return CART_LOAD_ERR_CREATE_SAVES_SUBDIR_FAIL;
-        }
-        path_push_subdir(path, SAVES_SUBDIRECTORY);
+    switch (save_type)
+    {
+        case SAVE_TYPE_SRAM:
+        case SAVE_TYPE_SRAM_128K:
+        case SAVE_TYPE_SRAM_BANKED:
+            path_ext_replace(path, "sra");
+            if (menu->settings.use_saves_folder) {
+                if ((save_type != FLASHCART_SAVE_TYPE_NONE) && create_saves_subdirectory(path)) {
+                    path_free(path);
+                    return CART_LOAD_ERR_CREATE_SAVES_SUBDIR_FAIL;
+                }
+                path_push_subdir(path, SAVES_SUBDIRECTORY);
+            }
+        break;
+
+        case SAVE_TYPE_EEPROM_4K:
+        case SAVE_TYPE_EEPROM_16K:
+            path_ext_replace(path, "eep");
+            if (menu->settings.use_saves_folder) {
+                if ((save_type != FLASHCART_SAVE_TYPE_NONE) && create_saves_subdirectory(path)) {
+                    path_free(path);
+                    return CART_LOAD_ERR_CREATE_SAVES_SUBDIR_FAIL;
+                }
+                path_push_subdir(path, SAVES_SUBDIRECTORY);
+            }
+        break;
+
+        case SAVE_TYPE_FLASHRAM:
+        case SAVE_TYPE_FLASHRAM_PKST2:
+            path_ext_replace(path, "fla");
+            if (menu->settings.use_saves_folder) {
+                if ((save_type != FLASHCART_SAVE_TYPE_NONE) && create_saves_subdirectory(path)) {
+                    path_free(path);
+                    return CART_LOAD_ERR_CREATE_SAVES_SUBDIR_FAIL;
+                }
+                path_push_subdir(path, SAVES_SUBDIRECTORY);
+            }
+        break;
+
+        default:
```

Comment:
Your change completely breaks existing `.sav` save files. I cannot accept this change in current state. Code path in `default` does essentially nothing because it is triggered on "none" save type only.

Please change the code to treat existing `.sav` files with priority. If no `.sav` file was found then try checking if `.sra`, `.eep`, or `.fla` save file exist. If that also fails then create a new save file with `.sav` extension.

I will add that your code has many copy-pasted segments that can be generalized. Try to optimize this.